src/main/java/thrift/storage/JsonAdaptedBudget.java

```diff
@@ -0,0 +1,71 @@
+package thrift.storage;
+
+import java.text.ParseException;
+import java.util.Calendar;
+import java.util.Date;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import thrift.commons.exceptions.IllegalValueException;
+import thrift.model.transaction.Budget;
+import thrift.model.transaction.BudgetValue;
+
+/**
+ * Jackson-friendly version of {@link Budget}.
+ */
+public class JsonAdaptedBudget {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Budget's %s field is missing!";
+
+    private final String period;
+    private final String value;
+
+    /**
+     * Constructs a {@code JsonAdaptedBudget} with the given budget details.
+     */
+    @JsonCreator
+    public JsonAdaptedBudget(@JsonProperty("period") String period, @JsonProperty("value") String value) {
+        this.period = period;
+        this.value = value;
+    }
+
+    /**
+     * Converts a given {@code Budget} into this class for Jackson use.
+     */
+    public JsonAdaptedBudget(Budget source) {
+        period = source.getBudgetDateString();
+        value = source.getBudgetValue().getUnformattedString();
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted budget object into the model's {@code Budget} object.
+     * @throws IllegalValueException if there are any data constraints violated in the adapted budget.
```

Comment:
There should be an empty line between description and parameter section.

Reply:
Fixed